httpcache.go

```diff
@@ -498,6 +516,41 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+// cachedReadCloser is a wrapper around ReadCloser R that calls OnEOF
+// handler with a full copy of the content read from that ReadCloser
+// when EOF is read.
+type cachedReadCloser struct {
```

Comment:
Minor nit, but I think `cachingReadCloser` might be a better name. It remembers the content from its underlying reader and makes it available at the end via `OnEOF`.

"cached" implies there was a cache available before we even started, but that's not the case.

Maybe `memorizingReadCloser` or so would be more descriptive, worth considering.